components/carbon-authenticators/saml2-sso-authenticator/org.wso2.carbon.identity.authenticator.saml2.sso.ui/src/main/resources/web/sso-acs/federation_ajaxprocessor.jsp

```diff
@@ -35,9 +35,7 @@
 
 	if (relayState!=null && !"null".equals(relayState)) {
     relayState = URLDecoder.decode(relayState, "UTF-8");
-    relayState = relayState.replaceAll("&", "&amp;").replaceAll("\"", "&quot;").replaceAll("'", "&apos;").
-            replaceAll("<", "&lt;").replaceAll(">", "&gt;").replace("\n", "");
```

Comment:
If we remove this I think it should break. We need to use the owasp library and encode for html in the form below directly where we replace. Then we can avoid this manual encoding.